modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackService.kt

```diff
@@ -396,44 +402,57 @@ open class PlaybackService : MediaBrowserServiceCompat(), CoroutineScope {
+                val filterEpisodes = playlistManager.findEpisodes(topFilter, episodeManager, playbackManager)
+                for (filterEpisode in filterEpisodes) {
+                    if (episodes.size >= NUM_SUGGESTED_ITEMS) {
+                        break
+                    }
+                    if (episodes.none { it.uuid == filterEpisode.uuid }) {
+                        episodes.add(filterEpisode)
+                    }
+                }
+            }
         }
-
-        Timber.d("Up next length was ${mediaUpNext.size}. Trying top filter.")
-        // If we don't have enough items in up next, try the top filter
-        val topPlaylist = playlistManager.findAll().firstOrNull()
-        if (topPlaylist == null) {
-            Timber.d("Could not find top filter.")
-            return ArrayList(mediaUpNext)
+        // add the latest episode
+        if (episodes.size < NUM_SUGGESTED_ITEMS) {
+            val latestEpisode = episodeManager.findLatestEpisodeToPlay()
+            if (latestEpisode != null && episodes.none { it.uuid == latestEpisode.uuid }) {
+                episodes.add(latestEpisode)
+            }
         }
-
-        Timber.d("Loading suggestions from ${topPlaylist.title}")
-        val playlistItems = loadEpisodeChildren(topPlaylist.uuid).take(NUM_SUGGESTED_ITEMS - mediaUpNext.size)
-        Timber.d("Got ${playlistItems.size} from playlist.")
-
-        val retList = mediaUpNext + playlistItems
-        Timber.d("Returning ${retList.size} suggestions. $retList")
-        return ArrayList(retList)
+        return convertEpisodesToMediaItems(episodes)
     }
 
-    private fun loadRecentChildren(): ArrayList<MediaBrowserCompat.MediaItem> {
+    suspend fun loadRecentChildren(): List<MediaBrowserCompat.MediaItem> {
         Timber.d("Loading recent children")
-        val upNext = playbackManager.getCurrentEpisode() ?: return arrayListOf()
-        val filesPodcast = Podcast(uuid = UserEpisodePodcastSubstitute.substituteUuid, title = UserEpisodePodcastSubstitute.substituteTitle)
-        val parentPodcast = (if (upNext is PodcastEpisode) podcastManager.findPodcastByUuid(upNext.podcastUuid) else filesPodcast) ?: return arrayListOf()
+        val episodes = listOfNotNull(upNextQueue.currentEpisode)
+        return convertEpisodesToMediaItems(episodes)
+    }
 
-        Timber.d("Recent item ${upNext.title}")
-        return arrayListOf(AutoConverter.convertEpisodeToMediaItem(this, upNext, parentPodcast))
+    private suspend fun convertEpisodesToMediaItems(episodes: List<BaseEpisode>): List<MediaBrowserCompat.MediaItem> {
+        return episodes.mapNotNull { episode ->
+            // find the podcast
+            val podcast = if (episode is PodcastEpisode) {
+                podcastManager.findPodcastByUuidSuspend(episode.podcastUuid)
+            } else {
+                Podcast(uuid = UserEpisodePodcastSubstitute.substituteUuid, title = UserEpisodePodcastSubstitute.substituteTitle)
```

Comment:
Not a change for this PR, but this got me thinking of whether we'd be better off having a global object along these lines instead of the `UserEpeisodePodcastSubstutite`:

``` kotlin
class Podcast {
  // ...
  companion object {
    // ...
    val userPodcast = Podcast(
      uuid = "da7aba5e-f11e-f11e-f11e-da7aba5ef11e",    // UserEpisodePodcastSubstitute.substituteUuid
      title = "Custom Episode",                         // UserEpisodePodcastSubstitute.substituteTitle
    )
  }
```

That would avoid us reconstructing this object multiple times (not that we do it that often), and would feel a bit more intuitive. I'd mildly prefer to make a UserEpisodePodcast `object` instead of just making it a value, but that's not possible since `Podcast` is a data object.

Anyway, let me know what you think because I might make that change at some point if you like it. I don't feel strongly though.

Reply:
Yes, that would be great! Also I can never remember `UserEpisodePodcastSubstitute` so it takes me a while to find it.